fix: Fixed training scripts

frgfm opened the pull request:
This PR aims at reducing the discrepancies between training run with fastai and bare pytorch. The later was running by default with fully unfrozen model while the first was only unfreezing the fully connected layers.

Also added checkpoint resuming option on both scripts, and a binary classification option on the pytorch script

frgfm committed:
fix: Fixed layer freezing

Pure pytorch reference script had the model fully unfrozen by default which would not be comparable to the one using fastai. See #3

frgfm committed:
feat: Added options for model unfreezing

frgfm committed:
style: Corrected whitespaces and indentation

frgfm committed:
refactor: Switched to TTA

Test Time Augmentation is now similar to training augmentation.

frgfm committed:
feat: Added resume option on references

blenzi commented on references/classification/fastai/train.py:
```diff
@@ -71,21 +75,20 @@ def main(args):
-        help="Use pre-trained models from the modelzoo",
-        action="store_true",
-    )
+    parser.add_argument('--resume', default=None, help='resume from checkpoint')
```
It's a detail but the way the help is written, the use may think it is a yes/no while the user has to provide the name of the checkpoint if I understand well.

frgfm committed:
style: Clarified argument description

frgfm committed:
fix: Moved saving message display

Saving message should only be displayed if the checkpoint is actually being created.

frgfm committed:
docs: Updated docstrings

frgfm committed:
feat: Added binary classification option

See #19

frgfm committed:
style: Fixed pep8 compliance

frgfm committed:
Merge branch 'master' into ref-fix

frgfm committed:
fix: Fixed argument description

frgfm committed:
fix: Fixed training script saving message

blenzi reviewed (approved):
Maybe you know this: in the future, in case you want to show the default arguments, you can use:
https://docs.python.org/3/library/argparse.html#argparse.ArgumentDefaultsHelpFormatter
